src/interceptors/response-time.cpu.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor, Logger } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { MetricsService, PerformanceProfiler } from '@multiversx/sdk-nestjs-monitoring';
+
+@Injectable()
+export class ResponseTimeCpuInterceptor implements NestInterceptor {
+  private readonly logger = new Logger(ResponseTimeCpuInterceptor.name);
+  private readonly shouldLog: boolean;
```

Comment:
we could have kept the entire `options` object as class member. but it can remain like this

Reply:
we can leave it like that